src/common/snippets/src/lowered/pass/propagate_subtensors.cpp

```diff
@@ -15,14 +15,43 @@ namespace snippets {
 namespace lowered {
 namespace pass {
 namespace {
+
+// The algorithm uses the following special values in subtensors/shapes:
+// 1. Dynamic value in subtensor/shape : SIZE_MAX
+// 2. Full fimension in subtensor      : SIZE_MAX - 1
+// 3. Default value of `new_dim_value` : SIZE_MAX - 2
+// 4. `Forced` special dynamic value   : SIZE_MAX - 3
+//
+// We have to introduce `SPECIAL_DYNAMIC_VALUE` to distinguish `new_dim_value = DYNAMIC`
```

Comment:
Firstly, thanks a lot for the detailed explanatory comment!
`SPECIAL_DYNAMIC_VALUE` on this line -- is that `FORCED_DYNAMIC_VALUE`, right? Could you please correct a comment please then?